ome_zarr/format.py

```diff
@@ -215,5 +241,60 @@ def validate_well_dict(
         if well["columnIndex"] != columns.index(column):
             raise ValueError(f"Mismatching column index for {well}")
 
+    def generate_coordinate_transformations(
+        self, shapes: List[tuple]
+    ) -> Optional[List[List[Dict[str, Any]]]]:
+
+        data_shape = shapes[0]
+        coordinate_transformations: List[List[Dict[str, Any]]] = []
+        # calculate minimal 'scale' transform based on pyramid dims
+        for shape in shapes:
+            assert len(shape) == len(data_shape)
+            scale = [full / level for full, level in zip(data_shape, shape)]
+            coordinate_transformations.append([{"type": "scale", "scale": scale}])
+
+        return coordinate_transformations
+
+    def validate_coordinate_transformations(
+        self,
+        shapes: List[tuple],
+        coordinate_transformations: List[List[Dict[str, Any]]] = None,
+    ) -> None:
+        """
+        Validates that a list of dicts contains a 'scale' transformation
+
+        Raises ValueError if no 'scale' found or doesn't match ndim
+        """
+
+        if coordinate_transformations is None:
+            raise ValueError("coordinate_transformations must be provided")
+        ct_count = len(coordinate_transformations)
+        if ct_count != len(shapes):
+            raise ValueError(
+                "coordinate_transformations count: %s must match datasets %s"
+                % (ct_count, len(shapes))
+            )
+        for shape, transformations in zip(shapes, coordinate_transformations):
+            ndim = len(shape)
+            assert isinstance(transformations, list)
+            scale_transfs = [
+                trans for trans in transformations if trans["type"] == "scale"
+            ]
+            if len(scale_transfs) != 1:
+                raise ValueError(
+                    "Must supply 1 'scale' item in coordinate_transformations"
```

Comment:
Two other potential elements to validate here:
- there should be at most one `translation`
- the `scale` transformation should be the first